ec2utils/ec2uploadimg/lib/ec2utils/ec2uploadimg.py

```diff
@@ -364,6 +372,23 @@ def _create_target_root_volume(self):
            the image we are creating."""
         return self._create_volume('%s' % self.root_volume_size)
 
+    # ---------------------------------------------------------------------
+    def _create_upload_key_pair(self, key_name='temporary_ec2_uploadkey'):
+        if self.verbose:
+            print('Creating temporary key pair')
+        home_dir = os.path.expanduser('~/')
+        fd, location = mkstemp(prefix='temporary_ec2_uploadkey.', suffix='.key', dir=home_dir)
```

Comment:
@rjschwei I didn't find a way to provide a TemporaryFile object. So it seems we need to create a file on the harddrive. Permissions of that file will be 600 and it's created in the user's home dir.